### ISSUE TITLE
Be able to create an AsyncHttpClient with a custom request timeout 

### ISSUE THREAD

**Issue by henricook** [http4s/http4s#3606]
I mentioned this briefly on Gitter and it was suggested I raise an issue.

I want to be able to create an AsyncHttpClient with a custom request timeout. At the moment I can use `AsyncHttpClient.resource[Task]()` to get the `defaultConfig` from http4s `AsyncHttpClient`. I might have missed something (please tell me if I have!) but I don't think I can tweak this to give me a custom request timeout.

`defaultConfig` is:

```scala
  val defaultConfig = new DefaultAsyncHttpClientConfig.Builder()
    .setMaxConnectionsPerHost(200)
    .setMaxConnections(400)
    .setRequestTimeout(defaults.RequestTimeout.toMillis.toInt)
    .setThreadFactory(threadFactory(name = { i =>
      s"http4s-async-http-client-worker-${i}"
    }))
    .setCookieStore(new NoOpCookieStore)
    .build()
```

I tried copying and pasting this into my own code but `NoOpCookieStore` is a package-private class and `threadFactory` was recently moved to http4s internals where I can't access it.

So my goal is to have one client with a 30 second timeout in use by my application (e.g. when talking to external services) but I want my acceptance test http4s client to have >30 seconds (e.g. 45 seconds) so it doesn't timeout whilst an external service being tested is timing out.

**TL;DR** - I just want to be able to set the request timeout for a client

Related:

@hamnis kindly pointed out that this is possible, which gives me an immediate workaround:

```scala
 AsyncHttpClient.resource({
      val b = new DefaultAsyncHttpClientConfig.Builder(AsyncHttpClient.defaultConfig)
      b.setRequestTimeout(3000)
      b.build()
    })
```

**Comment by hamnis** [comment]
We could add a function to make this easier.

`AsyncHttpClient.configure(fn: DefaultAsyncHttpClientConfig.Builder => DefaultAsyncHttpClientConfig.Builder): HttpClientConfig`

**Comment by jyoo980** [comment]
Could I try my hand at this?

I've use Scala at work but it'll be my first foray into Scala open-source.

**Comment by hamnis** [comment]
Fixed by #3623
